src/v/cluster/rm_stm.cc

```diff
@@ -319,17 +319,18 @@ ss::future<tx_errc> rm_stm::do_prepare_tx(
         co_return tx_errc::request_rejected;
     }
 
-    auto [preparing_it, _] = _mem_state.preparing.try_emplace(pid, tx_seq);
-    if (preparing_it->second != tx_seq) {
+    auto marker = prepare_marker{
+      .tm_partition = tm, .tx_seq = tx_seq, .pid = pid};
+    auto [_, inserted] = _mem_state.preparing.try_emplace(pid, marker);
+    if (!inserted) {
```

Comment:
It was the same before I just used an indirect way to check if try_emplace succeeds (by checking inserted value instead of looking on the `inserted` flag)